### PR TITLE
Use UnitControl for font-size

### PR DESCRIPTION
**Pull request by aristath**
## Description
This PR allows using non-px values for font-sizes, by using `UnitControl` instead of the custom number input we were previously using.
Font-sizes were previously saved using `px` as a hardcoded unit, so there are no backwards-compatibility issues I could find. This just allows using `em`/`rem` units.

## How has this been tested?
Tested in new blocks, tested in existing posts, nothing breaks. Tested using px, em & rem values. nothing breaks.

### Before:

Screenshot from 2021-04-29 10-39-17

### After:

Screenshot from 2021-04-29 10-52-53

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by carolinan** [comment]
I only have one question, doesn't this need to wait for
 _Continue to finish the FontSizePicker integration, i.e., finish migrating the components from #28399
 Re-introduce the FontSizePicker'_
In https://github.com/WordPress/gutenberg/issues/30503 ?

**Comment by aristath** [comment]
@carolinan tbh I don't know... Whatever the plan however, **font-sizes should not, under any circumstances, be limited to `px` values**. From what I've seen so far the g2 component doesn't support non-px values, so whether that gets implemented or not I don't know. Until the g2 component gets support for non-px units I think we'd be better off using the existing component which _can_ support em/rem units with this PR.
In other words, I'd say we implement it in the existing component, and if/when g2 gets support for other units, then we migrate the font-size to use that.

**Comment by aristath** [comment]
I see from the failing test there's an issue with zero values, I'll investigate further.

**Comment by sarayourfriend** [comment]
The G2 version of the font size picker is gone for now, to be later re-implemented once its dependencies are migrated. When it's re-implemented we can make sure that it uses UnitControl 👍 That way we don't introduce a regression.

**Review by sarayourfriend — COMMENTED** [review]
Code looks good to me but it appears that a couple tests are failing.

Another thing I noticed is that the label for `rem` is cut off: 
Captura de Tela 2021-04-29 às 07 08 17

**Comment by aristath** [comment]
> Another thing I noticed is that the label for `rem` is cut off

Yeah, that is the same for all `UnitControl` controls... so not related to this PR. For example this is the padding control for a group block:

Screenshot from 2021-04-30 08-15-19

**Comment by aristath** [comment]
All tests now pass :tada:

**Review by carolinan — APPROVED** [review]

File: packages/components/src/font-size-picker/index.js [hunk 1]
```diff
@@ -117,33 +131,20 @@ function FontSizePicker(
...
+						__unstableInputWidth="60px"
+						value={ value }
+						onChange={ ( nextSize ) => {
+							if ( 0 === parseFloat( nextSize ) || ! nextSize ) {
+								onChange( undefined );
+							} else {
+								onChange( nextSize );
+							}
+						} }
+						units={ CSS_UNITS }
```
Comment (youknowriad) [inline packages/components/src/font-size-picker/index.js:146]: I think themes can control the available units or disable them, is this being taken into consideration here?

**Comment by jasmussen** [comment]
This is cool. One thing I'm using on my own site, which isn't quite launched yet, is `vw` units for font-size. That allows some fluid typography:

vw

It might be fun to explore some [clamp](https://css-tricks.com/simplified-fluid-typography/) usage in the future to have some intrinsic responsiveness, but that's a thought for another day.

**Comment by aristath** [comment]
> This is cool. One thing I'm using on my own site, which isn't quite launched yet, is vw units for font-size. That allows some fluid typography

I do the same... but I do it for all elements of a page. So the `calc`, `vw`, `clamp` etc are on my root font-size, and then in elements I can use `em` to have adaptive typography. If I want something to be non-adaptive, I just use `px`.
I think youshould be able to use `vw` though if you want now... After https://github.com/WordPress/gutenberg/pull/31822 was merged the available units are no longer hardcoded and you can use any valid CSS unit from this list https://github.com/WordPress/gutenberg/pull/31822/files#diff-4161524a5c509693e7a2412feb7767fa2c69d841e870fc80a35d33ac3fbeb28cR17-R108. I believe I included all units there :wink:

**Comment by jasmussen** [comment]
Just to follow up, I explored query-less responsiveness a bit more in [this codepen](https://codepen.io/joen/pen/wvJMdGr?editors=1100). I also leveraged some of that for [my theme](https://github.com/jasmussen/mocco).
